lib/src/rules/null_closures.dart

```diff
@@ -0,0 +1,201 @@
+  new NonNullableFunction('dart.async', 'Timer', null, positional: [1]),
+  new NonNullableFunction('dart.async', 'Timer', 'periodic', positional: [1]),
+  new NonNullableFunction('dart.core', 'List', 'generate', positional: [1]),
+];
+
+List<NonNullableFunction> _staticFunctionsWithNonNullableArguments =
+    <NonNullableFunction>[
+  new NonNullableFunction('dart.async', null, 'scheduleMicrotask',
+      positional: [0]),
+  new NonNullableFunction('dart.async', 'Future', 'doWhile', positional: [0]),
+  new NonNullableFunction('dart.async', 'Future', 'forEach', positional: [1]),
+  new NonNullableFunction('dart.async', 'Future', 'wait', named: ['cleanUp']),
+  new NonNullableFunction('dart.async', 'Timer', 'run', positional: [0]),
+];
+
+List<NonNullableFunction> _instanceMethodsWithNonNullableArguments =
+    <NonNullableFunction>[
+  new NonNullableFunction('dart.async', 'Future', 'then',
+      positional: const [0], named: const ['onError']),
+  new NonNullableFunction('dart.async', 'Future', 'complete', positional: [0]),
+  new NonNullableFunction('dart.collection', 'Queue', 'removeWhere',
+      positional: [0]),
+  new NonNullableFunction('dart.collection', 'Queue', 'retainWhere',
+      positional: [0]),
+  new NonNullableFunction('dart.collection', 'ListQueue', 'forEach',
+      positional: [0]),
+  new NonNullableFunction('dart.collection', 'ListQueue', 'removeWhere',
+      positional: [0]),
+  new NonNullableFunction('dart.collection', 'ListQueue', 'retainWhere',
+      positional: [0]),
+  new NonNullableFunction('dart.collection', 'MapView', 'forEach',
+      positional: [0]),
+  new NonNullableFunction('dart.collection', 'MapView', 'putIfAbsent',
+      positional: [1]),
+  new NonNullableFunction('dart.core', 'List', 'any', positional: [0]),
+  new NonNullableFunction('dart.core', 'List', 'every', positional: [0]),
+  new NonNullableFunction('dart.core', 'List', 'expand', positional: [0]),
+  new NonNullableFunction('dart.core', 'List', 'firstWhere',
+      positional: const [0], named: const ['orElse']),
+  new NonNullableFunction('dart.core', 'List', 'fold', positional: [1]),
+  new NonNullableFunction('dart.core', 'List', 'forEach', positional: [0]),
+  new NonNullableFunction('dart.core', 'List', 'lastWhere',
+      positional: [0], named: ['orElse']),
+  new NonNullableFunction('dart.core', 'List', 'map', positional: [0]),
+  new NonNullableFunction('dart.core', 'List', 'reduce', positional: [0]),
+  new NonNullableFunction('dart.core', 'List', 'removeWhere', positional: [0]),
+  new NonNullableFunction('dart.core', 'List', 'retainWhere', positional: [0]),
+  new NonNullableFunction('dart.core', 'List', 'singleWhere', positional: [0]),
+  new NonNullableFunction('dart.core', 'List', 'skipWhile', positional: [0]),
+  new NonNullableFunction('dart.core', 'List', 'takeWhile', positional: [0]),
+  new NonNullableFunction('dart.core', 'List', 'where', positional: [0]),
```

Comment:
`Iterable` instead of `List` ?

Reply:
Use `DartTypeUtilities.implementsAnyInterface` ?

Reply:
Done! Great suggestion!